inception-ui-kb/src/main/java/de/tudarmstadt/ukp/inception/ui/kb/AnnotatedListIdentifiers.java

```diff
@@ -133,9 +134,11 @@ protected void onConfigure()
             currentProject = kbModel.getObject().getProject();
             return searchService.query(currentUser, currentProject, targetQuery.getObject());
         }
+        catch (ExecutionException e) {
```

Comment:
Uhm... that seems too generic a solution. Why is the ExecutionException thrown in the first place?

Reply:
Its from SearchService, i wanted to add InxNotFoundException but ExecutionException takes over that in SearchService

Reply:
By why does the search service generate this exception at all? Even if there are no documents, the search service should not generate exceptions when one does a query.

Reply:
It throws over from 
```results = index.getPhysicalIndex().executeQuery(aUser, aQuery, null, (String) null);```
 in ```query(User aUser, Project aProject, String aQuery)``` . The exception comes from 
`executeQuery(User aUser, String aQuery, String aSortOrder,
            String... aResultField)` in MtasDocumentIndex.java at line `IndexReader indexReader = DirectoryReader.open(directory);` . The idea may be that if in case a user fires a wrong query , it should return an error message .

Reply:
This is the case for IndexNotFoundException  with 

``` 
Caused by: org.apache.lucene.index.IndexNotFoundException: no segments* file found in MMapDirectory@/home/kumar/.inception/repository/project/10/indexMtas lockFactory=org.apache.lucene.store.NativeFSLockFactory@3a9fbd7a: files: [write.lock]
	at org.apache.lucene.index.SegmentInfos$FindSegmentsFile.run(SegmentInfos.java:665) ~[lucene-core-7.1.0.jar:7.1.0 84c90ad2c0218156c840e19a64d72b8a38550659 - ubuntu - 2017-10-13 16:12:42]
	at org.apache.lucene.index.StandardDirectoryReader.open(StandardDirectoryReader.java:79) ~[lucene-core-7.1.0.jar:7.1.0 84c90ad2c0218156c840e19a64d72b8a38550659 - ubuntu - 2017-10-13 16:12:42]
	at org.apache.lucene.index.DirectoryReader.open(DirectoryReader.java:63) ~[lucene-core-7.1.0.jar:7.1.0 84c90ad2c0218156c840e19a64d72b8a38550659 - ubuntu - 2017-10-13 16:12:42]
	at de.tudarmstadt.ukp.inception.search.index.mtas.MtasDocumentIndex.executeQuery(MtasDocumentIndex.java:170) ~[classes/:?]
	... 203 more
``` 
MtasDocumentIndex.java:170 throws ExecutionException()